lib/tooling/clang-query-tool.js

```diff
@@ -42,10 +42,12 @@ export class ClangQueryTool extends BaseTool {
             compileFlags.push(this.tool.options);
         }
 
+        const query_commands_file = this.getUniqueFilePrefix() + 'query_commands.txt';
+
         await fs.writeFile(path.join(dir, 'compile_flags.txt'), compileFlags.join('\n'));
```

Comment:
Do we not also have a race on this file here? If multiple tools are trying to write to it at the same time, then potentially we'll see incomplete data in it?